pkg/server/plugin/nodeattestor/azure/msi.go

```diff
@@ -0,0 +1,193 @@
+package azure
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/common/jwtutil"
+	"github.com/spiffe/spire/pkg/common/plugin/azure"
+	spi "github.com/spiffe/spire/proto/common/plugin"
+	"github.com/spiffe/spire/proto/server/nodeattestor"
+	"github.com/zeebo/errs"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+const (
+	pluginName = "azure_msi"
+
+	tokenLeeway           = time.Minute * 5
+	keySetRefreshInterval = time.Hour
+	azureConfigURL        = "https://login.microsoftonline.com/common/.well-known/openid-configuration"
```

Comment:
related to earlier discovery document comment: maybe we can make this opinionated towards oidc discovery. the `/.well-known/openid-configuration` part of the path is defined by spec. we could rename this to `azureOIDCIssuer` with a value of `https://login.microsoftonline.com/common/`

https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationRequest

Reply:
SGTM